macro/src/expand.rs

```diff
@@ -1985,14 +1983,13 @@ fn expand_shared_ptr(
         .explicit_impl
         .map_or(key.end_span, |explicit| explicit.brace_token.span.join());
     let unsafe_token = format_ident!("unsafe", span = begin_span);
-    let not_destructible_err = format!("{} is not destructible", display_namespaced(resolve.name));
 
     quote_spanned! {end_span=>
         #cfg
         #[automatically_derived]
-        #unsafe_token impl #impl_generics ::cxx::memory::SharedPtrTarget for #ident #ty_generics {
+        #unsafe_token impl #impl_generics ::cxx::memory::SharedPtrTarget for #inner #ty_generics {
             fn __typename(f: &mut ::cxx::core::fmt::Formatter<'_>) -> ::cxx::core::fmt::Result {
-                f.write_str(#name)
+                f.write_str(::core::stringify!(#inner))
```

Comment:
Using stringify for this is not a good approach.

```rust
#[cxx::bridge]
pub mod ffi {
    struct Context<'a> {
        session: &'a str,
    }
    struct Server<'srv> {
        ctx: SharedPtr<Context<'srv>>,
    }
    struct Client<'clt> {
        ctx: SharedPtr<Context<'clt>>,
    }
}

fn main() {
    let _ = *cxx::SharedPtr::<ffi::Context>::null();
}
```

Before:

```console
called deref on a null SharedPtr<Context>
```

After:

```console
called deref on a null SharedPtr<Context < 'srv >>
```

The spacing is wonky, and this is using a random one of the lifetime names which is going to be misleading for the user if their panic is happening in "client" code that has nothing to do with "server" code.